packages/web-media-box/src/hls-parser/tags/tagWithAttributesProcessors.ts

```diff
@@ -240,3 +240,51 @@ export class ExtXMedia extends TagWithAttributesProcessor {
     playlist.renditionGroups[renditionTypeKey][rendition.groupId] = [rendition];
   }
 }
+
+export class ExtXDaterange extends TagWithAttributesProcessor {
+  private static readonly ID = 'ID';
+  private static readonly CLASS = 'CLASS';
+  private static readonly START_DATE = 'START-DATE';
+  private static readonly CUE = 'CUE';
+  private static readonly END_DATE = 'END-DATE';
+  private static readonly DURATION = 'DURATION';
+  private static readonly PLANNED_DURATION = 'PLANNED-DURATION';
+  // Client attributes look like X-<client-attribute>, example: X-COM-EXAMPLE-AD-ID="XYZ123" 
+  private static readonly CLIENT_ATTRIBUTES = 'X-';
+  private static readonly SCTE35_CMD = 'SCTE35-CMD';
+  private static readonly SCTE35_OUT = 'SCTE35-OUT';
+  private static readonly SCTE35_IN = 'SCTE35-IN';
+  private static readonly END_ON_NEXT = 'END-ON-NEXT';
+
+  protected requiredAttributes = new Set([ExtXDaterange.ID, ExtXDaterange.START_DATE]);
+  protected tag = EXT_X_DATERANGE;
+
+  protected safeProcess(tagAttributes: Record<string, string>, playlist: ParsedPlaylist): void {
+    if (!playlist.dateRanges) {
+      playlist.dateRanges = [];
+    }
```

Comment:
you can move it to the playlist initializer and you can probably remove `?`  from `dateRanges?: DateRange[];` model

Reply:
Good call, fixed.